mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -113,6 +115,8 @@ bool MemRefDependenceGraph::init() {
   // Map from a memref to the set of ids of the nodes that have ops accessing
   // the memref.
   DenseMap<Value, SetVector<unsigned>> memrefAccesses;
+  // Vector of nodes with possible memoryEffects in order of traversal
+  SmallVector<Node *> memoryEffectNodes;
```

Comment:
This vector holds the for-nodes with potential memory effects in the order of traversal